### ISSUE TITLE
Percussion panel - additional keyboard navigation requirements

### ISSUE THREAD

**Issue by mathesoncalum** [musescore/MuseScore#26352]
### Overview

This issue outlines 3 additional requirements for the percussion panel in relation to keyboard navigation/accessibility.

---

### 1. Toolbar navigation

[Original report](https://github.com/musescore/MuseScore/pull/25810#issuecomment-2556010538) by @avvvvve:

> Pressing the left arrow when focusing within certain groups won't cycle back to the last focusable thing. i.e. focus on the Write/Preview button. Pressing the right, up, or down arrows over and over will swap between the two options, but pressing left when on "Write" does nothing. Same problem for 'Layout'/'Customize kit'.

Whoever investigates this (probably me) will need to consider whether this can/should be addressed at the percussion panel level, or if it's something that needs to be addressed at a higher level in our navigation system.

---

### 2. Pad navigation

[Original report](https://github.com/musescore/MuseScore/pull/25810#issuecomment-2556010538) by @avvvvve:

> Pressing up while focused in the middle of the top row moves focus left rather than back to the last row. Vice versa for the bottom row or a pad with empty slots beneath it: when pressing down, focus moves right when it should move down to the next row or back to the first one if you're already in the last row.

Same goes for this one, W.R.T. solving it at the navigation system level.

---

### 3. Screen reader behaviour (design needed)

As outlined [here](https://github.com/musescore/MuseScore/pull/25518#issuecomment-2519036578) by @shoogle, the screen reader should give some more detailed information when navigating pads (and footers). A design decision is needed here regarding the exact information to include.

### Checklist

- [x] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [x] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by avvvvve** [comment]
### Thoughts on point 3

#### 1. Naming column/row numbers
I can see it being helpful to know the row/column position, but potentially annoying that it might add comprehension time & effort to navigating through the pads. I'd say leave it in for now (it seems to already be implemented), but I wonder if we could get screenreaders to read it faster (i.e. without pauses) by removing punctuation from the syntax.

Like, instead of

- "Row: 3; Column: 3;"...
     <img width="240" alt="Image" src="https://github.com/user-attachments/assets/4d836c67-c02a-4662-b9ed-fd66318fd96e" />

We could do:

- "Row 3 Column 3"

^ @shoogle Please advise if this isn't good practice :) 

#### 2. Naming of the bottom part of the pad

I think we could use a more descriptive name for the bottom section of the pad than 'Footer'. Perhaps 'options'?

<img width="506" alt="Image" src="https://github.com/user-attachments/assets/8b72a3df-9171-4e43-af41-5e58634fed39" />

#### 3. Where to read the shortcut + MIDI assignment

I'm tempted to say we should read the shortcut/MIDI assignment with the main pad, not the options button (even though they're on-screen in the options button). I think since many pads won't come with a default shortcut, we should _not_ read "No shortcut" when this is the case.

#### So to bring it all together...

Image | ![image](https://github.com/user-attachments/assets/fad0a49d-f0f0-4b0e-965d-6db2eccb9634) | ![image](https://github.com/user-attachments/assets/94515261-987c-4e0b-b24d-b65154f88f57)
:---|:---:|:---:
Pad speech: | _Bass Drum 1 pad button, Shortcut B, MIDI C2, Row 3 Column 3_ | _Side Stick pad button, MIDI C sharp 2, Row 1 Column 5_
Footer speech: | _Bass Drum 1 options button_ | _Side Stick options button_

Open to suggestions on that approach!

-----

Also, I noticed this when using VoiceOver on Mac (not sure if it's specific to that screenreader or not)

### 4. Dialog reads additional text outside dialog
I had VoiceOver on and was editing the layout of the percussion panel. I dragged a panel to a different slot, and when the MIDI preference dialog came up, it read the title, but then continued to read some parts of the Properties panel which went on for a long time.

https://github.com/user-attachments/assets/7062dd31-7d99-4bcf-a757-f8ec8b21fdf0

**Comment by avvvvve** [comment]
@mathesoncalum just checking, should this one stay in 4.5 since there are still unfinished parts? (I'm assuming the automations just did that)

**Comment by mathesoncalum** [comment]
Sorry should have mentioned @avvvvve - after discussing privately with @bkunda we agreed that only point 3 needed to be addressed for 4.5. I've put this in the "further future" backlog for now but feel free to change.

**Comment by jmwmusic** [comment]
In regard to number 2.  The navigation is also incorrect when at the end of a row.  If in the last column you press right you would expect it to cycle back to the first item of that same row, instead pressing right moves down to the last item of the next row, so effectively right is working like down at the end of a row.

In many instances simply having all flow through from the end of 1 row to the start of the next with right makes sense.  But in this case as it is very much designed as a grid and those grid numbers are also announced and can be customised it makes more sense to cycle through each row only with left/right and the user should use up/down to move in that way, those also should cycle through each row, so if on the last row you should go back to the first one with another down arrow.  This is how a grid should work.  If you have a commonly used item at the end of the row and want to get back to the drum at the start it is much quicker to press right arrow once more to start again than to have to arrow left through 8 items.  Quite different to palettes where instead of multiple moves around the one area for multiple selection you are just skimming right through to find 1 item.
